Objects/abstract.c

```diff
@@ -1421,17 +1423,17 @@ _PyNumber_Index(PyObject *item)
 
     if (!PyLong_Check(result)) {
         PyErr_Format(PyExc_TypeError,
-                     "__index__ returned non-int (type %.200s)",
```

Comment:
I don't see much reason to change this error message either.

Reply:
I changed it to the more common way as you recommended elsewhere: `__method__() must return a int, not ...`
and added type info before method name, like for other methods